src/main/java/no/nav/pto/veilarbportefolje/persononinfo/barnUnder18Aar/BarnUnder18AarService.java

```diff
@@ -49,25 +52,32 @@ public List<Fnr> hentBarnFnrsForForeldre(List<Fnr> fnrForeldre) {
         return result;
     }
 
-    public void lagreBarnOgForeldreansvar(Fnr foresattIdent, List<BarnUnder18Aar> barnFraPdl) {
-        List<Fnr> lagredeBarn = barnUnder18AarRepository.hentForeldreansvarForPerson(foresattIdent);
+    @Transactional
+    public void lagreBarnOgForeldreansvar(Fnr foresattIdent, List<Fnr> foreldreAnsvar) {
+        try {
+            List<Fnr> lagredeBarn = barnUnder18AarRepository.hentForeldreansvarForPerson(foresattIdent);
 
-        List<Fnr> barnFnrFraPdl = barnFraPdl.stream().map(BarnUnder18Aar::getFnr).toList();
-
-        lagredeBarn.forEach(barnFnr -> {
-            if (!barnFnrFraPdl.contains(barnFnr)) {
-                barnUnder18AarRepository.slettForeldreansvar(foresattIdent, barnFnr);
-                slettBarnDataHvisIngenForeldreErUnderOppfolging(barnFnr);
-                secureLog.warn(String.format("Barn fjernet fra PDL for foreldre %s og barn %s", foresattIdent, barnFnr));
-            }
-        });
-
-        barnFraPdl.forEach(barnUnder18Aar -> {
-            if (erUnder18Aar(barnUnder18Aar.getFodselsdato())) {
-                barnUnder18AarRepository.lagreBarnData(barnUnder18Aar.getFnr(), barnUnder18Aar.getFodselsdato(), barnUnder18Aar.getDiskresjonskode());
-                barnUnder18AarRepository.lagreForeldreansvar(foresattIdent, barnUnder18Aar.getFnr());
-            }
-        });
+            lagredeBarn.forEach(barnFnr -> {
+                if (!foreldreAnsvar.contains(barnFnr)) {
```

Comment:
can you rename foreldreansvar to foreldreansvarFraPDL?

Reply:
Yep